networking_ccloud/common/config/config_driver.py

```diff
@@ -431,6 +431,29 @@ class GlobalConfig(pydantic.BaseModel):
     _normalize_vlan_ranges = pydantic.validator('default_vlan_ranges',
                                                 each_item=True, allow_reuse=True)(validate_vlan_ranges)
 
+    @pydantic.root_validator
+    def check_vrf_name_unique(cls, values):
+        if 'vrfs' not in values:
+            return values
+        names = set()
+        for vrf in values['vrfs']:
+            if vrf.name in names:
+                raise ValueError(f'VRF {vrf.name} is duplicated')
+            else:
+                names.add(vrf.name)
+        return values
+
+    @pydantic.root_validator
+    def check_vrf_number_unique(cls, values):
+        if 'vrfs' not in values:
+            return values
+        nums = set()
+        for vrf in values['vrfs']:
+            if vrf.number in nums:
+                raise ValueError(f'VRF id {vrf.number} is duplicated on VRF {vrf.name}')
+            else:
+                nums.add(vrf.number)
+        return values
```

Comment:
If these two validators only do validation on the `vrf` attribute they don't need to be `root_validators`, do they? Could make them to be a validator only for `vrf`.

Reply:
They do the validation on the list of VRFs, so they must be root validators, don't they?

Reply:
The `vrfs` field is a single field / member of the class `GlobalConfig`. If you write a validator for it you can tell it to just validate `vrfs` and it will get the whole list (as long as you don't provide `each_item=True` to the validator).